flink-libraries/flink-ml/src/main/scala/org/apache/flink/ml/recommendation/ALS.scala

```diff
@@ -273,6 +308,14 @@ object ALS {
     val defaultValue: Option[Int] = Some(10)
   }
 
+  case object ImplicitPrefs extends Parameter[Boolean] {
```

Comment:
Can't find a way to comment on line 264/299 but we should take the opportunity to set the default number of factors to a more reasonable 50, and add to the docstring and documentation the recommendation:

> we recommend working with the highest number of factors feasible within computational limitations.

Which comes straight from the iALS paper.